nexus/db-queries/src/db/datastore/switch_port.rs

```diff
@@ -312,319 +316,2021 @@ impl DataStore {
+                                .map_err(|e: diesel::result::Error| {
+                                    match e {
+                                        diesel::result::Error::NotFound => {
+                                            err.bail(Error::not_found_by_id(ResourceType::BgpConfig, &id))
+                                        },
+                                        _ => {
+                                            let message = "error while looking up bgp config for bgp peer";
+                                            error!(opctx.log, "{message}"; "error" => ?e);
+                                            err.bail(Error::internal_error(message))
+                                        },
+                                    }
+                                })
+                        },
+
+                        NameOrId::Name(name) => {
+                            bgp_config::table
+                                .filter(bgp_config::time_deleted.is_null())
+                                .filter(bgp_config::name.eq(name.to_string()))
+                                .select(bgp_config::id)
+                                .limit(1)
+                                .first_async::<Uuid>(&conn)
+                                .await
+                                .map_err(|e: diesel::result::Error| {
+                                    match e {
+                                        diesel::result::Error::NotFound => {
+                                            err.bail(Error::not_found_by_name(ResourceType::BgpConfig, &name))
+                                        },
+                                        _ => {
+                                            let message = "error while looking up bgp config for bgp peer";
+                                            error!(opctx.log, "{message}"; "error" => ?e);
+                                            err.bail(Error::internal_error(message))
+                                        },
+                                    }
+                                })
+                        }
+                    }?;
+
+                    let bgp_peer_config = SwitchPortBgpPeerConfig {
+                        port_settings_id,
+                        bgp_config_id,
+                        interface_name: new_settings.interface_name,
+                        addr: new_settings.addr.into(),
+                        hold_time: new_settings.hold_time.into(),
+                        idle_hold_time: new_settings.idle_hold_time.into(),
+                        delay_open: new_settings.delay_open.into(),
+                        connect_retry: new_settings.connect_retry.into(),
+                        keepalive: new_settings.keepalive.into(),
+                        remote_asn: new_settings.remote_asn.map(Into::into),
+                        min_ttl: new_settings.min_ttl.map(Into::into),
+                        md5_auth_key: new_settings.md5_auth_key,
```

Comment:
Is a clone required?

Also check line 638.

Reply:
Apologies, I am unsure of what you are referring to?

Reply:
Oops, I should have noticed/clarified that line 638 was from a different file that has the same name but different directory: nexus/db-model/src/switch_port.rs:638

For example, nexus/db-queries/src/db/datastore/switch_port.rs:590 has code that looks like:
```
md5_auth_key: p.md5_auth_key.clone(),
```

I was raising the question of, for any line of code that has a style of `md5_auth_key: xyz.md5_auth_key` (includes this line here for this comment, and above other file line 638), whether to consider if it should actually be `md5_auth_key: xyz.md5_auth_key.clone()`, because I've seen usage of `clone()` for lines of code related to `md5_auth_key` in both the `omicron` and `oxide.rs` repositories in the past.
See:
- https://github.com/oxidecomputer/oxide.rs/blob/e3e972bdc0d8d49c21b6b12c84b90c4d7a30eafc/cli/src/cmd_net.rs#L499
- https://github.com/oxidecomputer/oxide.rs/blob/e3e972bdc0d8d49c21b6b12c84b90c4d7a30eafc/cli/src/cmd_net.rs#L1097

This was part of my quick checking for consistency throughout the code that I looked at.

Reply:
Ah, ok! So generally `clone()` is required for any data you're "giving up ownership of" due to passing it to a different scope (like providing it as a parameter to a function), but you need to keep a copy of the valu in the current scope for use later on. So if I wanted to pass that value to multiple function calls, I would need to `clone` it. Here, we don't need to hold on to this data, so it's okay that we give our only "copy" of it to the function that we pass the struct to.

Fortunately, if you ever miss a `clone()` the code will fail to compile due to violating ownership / borrow checker rules.